scripts/partition_manager.py

```diff
@@ -687,6 +705,39 @@ def test():
     expect_addr_size(td, 'app', 600, 200)  # s spans a, b and c
     expect_addr_size(td, 'e', 800, 200)  # s spans a, b and c
 
+    # Verify that all 'share_size' with value partition that has size 0 is compatible withe 'one_of' dicts
+    td = {
+        'a': {'placement': {'after': 'start'}, 'size': 0},
+        'b': {'placement': {'after': {'one_of': ['a', 'start']}},
+              'share_size': ['a']},
+        'c': {'placement': {'after': {'one_of': ['a', 'b', 'start']}},
+              'share_size': {'one_of': ['a', 'b']}},
+        'd': {'placement': {'after': {'one_of': ['a', 'b', 'c', 'start']}},
+              'share_size': {'one_of': ['a', 'b', 'c']}},
+        # You get the point
+        'e': {'placement': {'after': {'one_of': ['a', 'b', 'c', 'd', 'start']}}, 'size': 100}
+    }
+    remove_all_zero_sized_partitions(td)
```

Comment:
Should there be an `assert 'a' not in td` here?

If not, why is `a` special just because it's the first partition? I am asking because `a` has zero size, and this function isn't named `remove_zero_sized_partitions_except_the_first_one()`.

Reply:
I will add the assert. I was too focused on verifying that the partitions which directly or indirectly shared size with a 0 sized partition was removed.